resources/recursor_install_rhel.rb

```diff
@@ -55,6 +56,7 @@
 
   yum_package 'pdns-recursor' do
     version new_resource.version
+    action new_resource.allow_upgrade ? :upgrade : :install
```

Comment:
Since the default action of `yum_package` is `:install` you can simplify this ternary with

```ruby
action :upgrade if new_resource.allow_upgrade
```